internal/persistence/persistence.go

```diff
@@ -21,31 +21,38 @@ import (
 
 	"github.com/hyperledger/firefly-common/pkg/fftypes"
 	"github.com/hyperledger/firefly-transaction-manager/pkg/apitypes"
+	"github.com/hyperledger/firefly-transaction-manager/pkg/toolkit"
 )
 
+// Persistence interface contains all the functions a persistence instance needs to implement.
+// Sub set of functions are grouped into sub interfaces to provide a clear view of what
+// persistent functions will be made available for each sub components to use after the persistent
+// instance is initialized by the manager.
 type Persistence interface {
+	EventStreamPersistence
+	ListenerPersistence
+	TransactionPersistence
+
+	// close function is controlled by the manager
+	Close(ctx context.Context)
+}
+
+type EventStreamPersistence interface {
 	WriteCheckpoint(ctx context.Context, checkpoint *apitypes.EventStreamCheckpoint) error
 	GetCheckpoint(ctx context.Context, streamID *fftypes.UUID) (*apitypes.EventStreamCheckpoint, error)
 	DeleteCheckpoint(ctx context.Context, streamID *fftypes.UUID) error
 
-	ListStreams(ctx context.Context, after *fftypes.UUID, limit int, dir SortDirection) ([]*apitypes.EventStream, error) // reverse UUIDv1 order
+	ListStreams(ctx context.Context, after *fftypes.UUID, limit int, dir toolkit.SortDirection) ([]*apitypes.EventStream, error) // reverse UUIDv1 order
 	GetStream(ctx context.Context, streamID *fftypes.UUID) (*apitypes.EventStream, error)
 	WriteStream(ctx context.Context, spec *apitypes.EventStream) error
 	DeleteStream(ctx context.Context, streamID *fftypes.UUID) error
-
-	ListListeners(ctx context.Context, after *fftypes.UUID, limit int, dir SortDirection) ([]*apitypes.Listener, error) // reverse UUIDv1 order
-	ListStreamListeners(ctx context.Context, after *fftypes.UUID, limit int, dir SortDirection, streamID *fftypes.UUID) ([]*apitypes.Listener, error)
+}
+type ListenerPersistence interface {
+	ListListeners(ctx context.Context, after *fftypes.UUID, limit int, dir toolkit.SortDirection) ([]*apitypes.Listener, error) // reverse UUIDv1 order
+	ListStreamListeners(ctx context.Context, after *fftypes.UUID, limit int, dir toolkit.SortDirection, streamID *fftypes.UUID) ([]*apitypes.Listener, error)
 	GetListener(ctx context.Context, listenerID *fftypes.UUID) (*apitypes.Listener, error)
 	WriteListener(ctx context.Context, spec *apitypes.Listener) error
 	DeleteListener(ctx context.Context, listenerID *fftypes.UUID) error
-
-	ListTransactionsByCreateTime(ctx context.Context, after *apitypes.ManagedTX, limit int, dir SortDirection) ([]*apitypes.ManagedTX, error)         // reverse create time order
```

Comment:
I'm sorry @Chengxuan - I'm not understanding why we're deleting these functions.

I believe what we need here is a `TransactionPersistence` interface here, which is the interface passed to the TX Handler, and implemented by the persistence implementation.

Reply:
I feel like I'm doing a bad job of explaining in english sorry, so I thought maybe a PR would be easier as it's a very straight forward change in the code itself: https://github.com/kaleido-io/firefly-transaction-manager/pull/2

Reply:
You see how now `toolkit` is dependent on `persistence`, and `txhandler`s are dependent on `toolkit`... but now `leveldb_persistence` is *not* dependent on `toolkit` (which was what seemed wrong to me).

Reply:
Hi @peterbroadhurst , thanks for the PR. I tried that initially but I couldn't make it work when I tired to write a transaction handler outside this repo. Becuase any logic there won't be able to reference a package in the `internal` folder in this repo, I believe this is a go constraint we shouldn't attempt to break. Therefore, there is no way to use the internal `TransactionPersistence` in any part of the code. Therefore, I did the otherwway around.

Reply:
I also did a similar externalization for the metrics interface: https://github.com/hyperledger/firefly-transaction-manager/pull/62/files#diff-c296a375f36621fa95cf261041d401ae007547bc96e5d2df9d01c07af6dd180c